### ISSUE TITLE
"ValueError: setting an array element with a sequence" when passing cutoff time DF to dfs()

### ISSUE THREAD

**Issue by maxpagels** [FeatureLabs/featuretools#362]
I have the following cutoff time dataframe:

```
instance_id | time | label
12443 | 2018-10-21 | 0
42354 | 2018-1-11 | 0
6546546 | 2018-2-2 | 0
56654 | 2018-10-2 | 0
3442 | 2018-1-18 | 0
234342 | 2018-12-15 | 0
4324 | 2018-4-17 | 1
```

With the following dtypes:

```
Out[19]:
instance_id            object
time           datetime64[ns]
label                   int64
```

Yet, when I pass the DF to dfs():

```
feature_matrix, feature_defs = ft.dfs(
    entityset=es,
    target_entity="customers",
    n_jobs=1,
    verbose=1,
    cutoff_time=ct
)
```

I get `ValueError: setting an array element with a sequence.`. The feature calculation works fine when not using cutoff_time. Is this a bug or am I passing in the wrong kind of dataframe?

Full trace below.

---

```
Built 298 features
Elapsed: 00:00 | Remaining: ? | Progress:   0%|          | Calculated: 0/10 chunks
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-867963ba8f1b> in <module>
      5     n_jobs=1,
      6     verbose=1,
----> 7     cutoff_time=ct
      8 )

/usr/local/lib/python3.7/site-packages/featuretools/synthesis/dfs.py in dfs(entities, relationships, entityset, target_entity, cutoff_time, instance_ids, agg_primitives, trans_primitives, allowed_paths, max_depth, ignore_entities, ignore_variables, seed_features, drop_contains, drop_exact, where_primitives, max_features, cutoff_time_in_index, save_progress, features_only, training_window, approximate, chunk_size, n_jobs, dask_kwargs, verbose)
    198                                                   n_jobs=n_jobs,
    199                                                   dask_kwargs=dask_kwargs,
--> 200                                                   verbose=verbose)
    201     else:
    202         feature_matrix = calculate_feature_matrix(features,

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in calculate_feature_matrix(features, entityset, cutoff_time, instance_ids, entities, relationships, cutoff_time_in_index, training_window, approximate, save_progress, verbose, chunk_size, n_jobs, dask_kwargs, profile)
    260                                                  cutoff_df_time_var=cutoff_df_time_var,
    261                                                  target_time=target_time,
--> 262                                                  pass_columns=pass_columns)
    263 
    264     feature_matrix = pd.concat(feature_matrix)

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in linear_calculate_chunks(chunks, features, approximate, training_window, profile, verbose, save_progress, entityset, no_unapproximated_aggs, cutoff_df_time_var, target_time, pass_columns)
    523                                           cutoff_df_time_var,
    524                                           target_time, pass_columns,
--> 525                                           backend=backend)
    526         feature_matrix.append(_feature_matrix)
    527         # Do a manual garbage collection in case objects from calculate_chunk

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in calculate_chunk(chunk, features, approximate, training_window, profile, verbose, save_progress, no_unapproximated_aggs, cutoff_df_time_var, target_time, pass_columns, backend, entityset)
    344                                            ids,
    345                                            precalculated_features=precalculated_features,
--> 346                                            training_window=window)
    347 
    348             id_name = _feature_matrix.index.name

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/utils.py in wrapped(*args, **kwargs)
     32         def wrapped(*args, **kwargs):
     33             if save_progress is None:
---> 34                 r = method(*args, **kwargs)
     35             else:
     36                 time = args[0].to_pydatetime()

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/calculate_feature_matrix.py in calc_results(time_last, ids, precalculated_features, training_window)
    318                                                     precalculated_features=precalculated_features,
    319                                                     ignored=all_approx_feature_set,
--> 320                                                     profile=profile)
    321             return matrix
    322 

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/pandas_backend.py in calculate_all_features(self, instance_ids, time_last, training_window, profile, precalculated_features, ignored, verbose)
    191 
    192                     handler = self._feature_type_handler(test_feature)
--> 193                     result_frame = handler(group, input_frames)
    194 
    195                     output_frames_type = self.feature_tree.output_frames_type(test_feature)

/usr/local/lib/python3.7/site-packages/featuretools/computational_backends/pandas_backend.py in _calculate_agg_features(self, features, entity_frames)
    479             nulls = pd.isnull(frame[f.get_name()])
    480             for ni in nulls[nulls].index:
--> 481                 frame.at[ni, f.get_name()] = f.default_value
    482 
    483         # 2. handle scalars default values

/usr/local/lib/python3.7/site-packages/pandas/core/indexing.py in __setitem__(self, key, value)
   2157         key = list(self._convert_key(key, is_setter=True))
   2158         key.append(value)
-> 2159         self.obj._set_value(*key, takeable=self._takeable)
   2160 
   2161 

/usr/local/lib/python3.7/site-packages/pandas/core/frame.py in _set_value(self, index, col, value, takeable)
   2580             series = self._get_item_cache(col)
   2581             engine = self.index._engine
-> 2582             engine.set_value(series._values, index, value)
   2583             return self
   2584         except (KeyError, TypeError):

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.set_value()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.set_value()

pandas/_libs/src/util.pxd in util.set_value_at()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

ValueError: setting an array element with a sequence.
```

**Comment by kmax12** [comment]
@maxpagels are you only using the default primitives? if you are using a primitive like NMostCommon, can you try to remove that and see if that fixes the problem.

**Comment by maxpagels** [comment]
@kmax12 good point, using only the default primitives seems to work. Is there a way of determining valid primitives beforehand?

**Comment by kmax12** [comment]
This is an issue with that particular primitive. All primitives should be valid. It will be fixed in the next release. I’ll keep this issue open until it’s fixed. Thanks letting us know.

**Comment by kmax12** [comment]
@maxpagels we just put out featuretools v0.6.0. can you upgrade and let us know if the issue has been resolved?

**Comment by rwedge** [comment]
Should be fixed as of v0.6.0
